File: AntlrLeftRecursionBenchmark/Python/LeftRecursionGrammarLexer.py
from antlr4 import *
from io import StringIO
from typing.io import TextIO
import sys



def serializedATN():
    with StringIO() as buf:
        buf.write("\3\u608b\ua72a\u8133\ub9ed\u417c\u3be7\u7786\u5964\2\21")
        buf.write("M\b\1\4\2\t\2\4\3\t\3\4\4\t\4\4\5\t\5\4\6\t\6\4\7\t\7")
        buf.write("\4\b\t\b\4\t\t\t\4\n\t\n\4\13\t\13\4\f\t\f\4\r\t\r\4\16")
        buf.write("\t\16\4\17\t\17\4\20\t\20\3\2\3\2\3\3\3\3\3\4\3\4\3\5")
        buf.write("\3\5\3\6\3\6\3\7\3\7\3\b\3\b\3\b\3\t\3\t\3\t\3\n\3\n\3")
        buf.write("\n\3\13\3\13\3\13\3\f\3\f\3\r\3\r\3\16\3\16\7\16@\n\16")
        buf.write("\f\16\16\16C\13\16\3\17\6\17F\n\17\r\17\16\17G\3\20\3")
        buf.write("\20\3\20\3\20\2\2\21\3\3\5\4\7\5\t\6\13\7\r\b\17\t\21")
        buf.write("\n\23\13\25\f\27\r\31\16\33\17\35\20\37\21\3\2\6\5\2C")
        buf.write("\\aac|\6\2\62;C\\aac|\3\2\62;\5\2\13\f\17\17\"\"\2N\2")
        buf.write("\3\3\2\2\2\2\5\3\2\2\2\2\7\3\2\2\2\2\t\3\2\2\2\2\13\3")
        buf.write("\2\2\2\2\r\3\2\2\2\2\17\3\2\2\2\2\21\3\2\2\2\2\23\3\2")
        buf.write("\2\2\2\25\3\2\2\2\2\27\3\2\2\2\2\31\3\2\2\2\2\33\3\2\2")
        buf.write("\2\2\35\3\2\2\2\2\37\3\2\2\2\3!\3\2\2\2\5#\3\2\2\2\7%")
        buf.write("\3\2\2\2\t\'\3\2\2\2\13)\3\2\2\2\r+\3\2\2\2\17-\3\2\2")
        buf.write("\2\21\60\3\2\2\2\23\63\3\2\2\2\25\66\3\2\2\2\279\3\2\2")
        buf.write("\2\31;\3\2\2\2\33=\3\2\2\2\35E\3\2\2\2\37I\3\2\2\2!\"")
        buf.write("\7`\2\2\"\4\3\2\2\2#$\7,\2\2$\6\3\2\2\2%&\7\61\2\2&\b")
        buf.write("\3\2\2\2\'(\7\'\2\2(\n\3\2\2\2)*\7-\2\2*\f\3\2\2\2+,\7")
        buf.write("/\2\2,\16\3\2\2\2-.\7?\2\2./\7?\2\2/\20\3\2\2\2\60\61")
        buf.write("\7#\2\2\61\62\7?\2\2\62\22\3\2\2\2\63\64\7(\2\2\64\65")
        buf.write("\7(\2\2\65\24\3\2\2\2\66\67\7~\2\2\678\7~\2\28\26\3\2")
        buf.write("\2\29:\7A\2\2:\30\3\2\2\2;<\7<\2\2<\32\3\2\2\2=A\t\2\2")
        buf.write("\2>@\t\3\2\2?>\3\2\2\2@C\3\2\2\2A?\3\2\2\2AB\3\2\2\2B")
        buf.write("\34\3\2\2\2CA\3\2\2\2DF\t\4\2\2ED\3\2\2\2FG\3\2\2\2GE")
        buf.write("\3\2\2\2GH\3\2\2\2H\36\3\2\2\2IJ\t\5\2\2JK\3\2\2\2KL\b")
        buf.write("\20\2\2L \3\2\2\2\5\2AG\3\2\3\2")
        return buf.getvalue()


class LeftRecursionGrammarLexer(Lexer):

    atn = ATNDeserializer().deserialize(serializedATN())

    decisionsToDFA = [ DFA(ds, i) for i, ds in enumerate(atn.decisionToState) ]

    CARET = 1
    STAR = 2
    SLASH = 3
    PERCENT = 4
    PLUS = 5
    MINUS = 6
    EQUAL = 7
    NOT_EQUAL = 8
    LOGICAL_AND = 9
    LOGICAL_OR = 10
    QUESTION = 11
    COLON = 12
    ID = 13
    DIGIT = 14
    WS = 15

    channelNames = [ u"DEFAULT_TOKEN_CHANNEL", u"HIDDEN" ]

    modeNames = [ "DEFAULT_MODE" ]

    literalNames = [ "<INVALID>",
            "'^'", "'*'", "'/'", "'%'", "'+'", "'-'", "'=='", "'!='", "'&&'",
            "'||'", "'?'", "':'" ]

    symbolicNames = [ "<INVALID>",
            "CARET", "STAR", "SLASH", "PERCENT", "PLUS", "MINUS", "EQUAL",
            "NOT_EQUAL", "LOGICAL_AND", "LOGICAL_OR", "QUESTION", "COLON",
            "ID", "DIGIT", "WS" ]

    ruleNames = [ "CARET", "STAR", "SLASH", "PERCENT", "PLUS", "MINUS",
                  "EQUAL", "NOT_EQUAL", "LOGICAL_AND", "LOGICAL_OR", "QUESTION",
                  "COLON", "ID", "DIGIT", "WS" ]

    grammarFileName = "LeftRecursionGrammar.g4"

    def __init__(self, input=None, output:TextIO = sys.stdout):
        super().__init__(input, output)
        self.checkVersion("4.8")
        self._interp = LexerATNSimulator(self, self.atn, self.decisionsToDFA, PredictionContextCache())
        self._actions = None
        self._predicates = None

File: AntlrLeftRecursionBenchmark/JavaScript/LeftRecursionGrammarLexer.js
// jshint ignore: start
var antlr4 = require('antlr4/index');



var serializedATN = ["\u0003\u608b\ua72a\u8133\ub9ed\u417c\u3be7\u7786\u5964",
    "\u0002\u0011M\b\u0001\u0004\u0002\t\u0002\u0004\u0003\t\u0003\u0004",
    "\u0004\t\u0004\u0004\u0005\t\u0005\u0004\u0006\t\u0006\u0004\u0007\t",
    "\u0007\u0004\b\t\b\u0004\t\t\t\u0004\n\t\n\u0004\u000b\t\u000b\u0004",
    "\f\t\f\u0004\r\t\r\u0004\u000e\t\u000e\u0004\u000f\t\u000f\u0004\u0010",
    "\t\u0010\u0003\u0002\u0003\u0002\u0003\u0003\u0003\u0003\u0003\u0004",
    "\u0003\u0004\u0003\u0005\u0003\u0005\u0003\u0006\u0003\u0006\u0003\u0007",
    "\u0003\u0007\u0003\b\u0003\b\u0003\b\u0003\t\u0003\t\u0003\t\u0003\n",
    "\u0003\n\u0003\n\u0003\u000b\u0003\u000b\u0003\u000b\u0003\f\u0003\f",
    "\u0003\r\u0003\r\u0003\u000e\u0003\u000e\u0007\u000e@\n\u000e\f\u000e",
    "\u000e\u000eC\u000b\u000e\u0003\u000f\u0006\u000fF\n\u000f\r\u000f\u000e",
    "\u000fG\u0003\u0010\u0003\u0010\u0003\u0010\u0003\u0010\u0002\u0002",
    "\u0011\u0003\u0003\u0005\u0004\u0007\u0005\t\u0006\u000b\u0007\r\b\u000f",
    "\t\u0011\n\u0013\u000b\u0015\f\u0017\r\u0019\u000e\u001b\u000f\u001d",
    "\u0010\u001f\u0011\u0003\u0002\u0006\u0005\u0002C\\aac|\u0006\u0002",
    "2;C\\aac|\u0003\u00022;\u0005\u0002\u000b\f\u000f\u000f\"\"\u0002N\u0002",
    "\u0003\u0003\u0002\u0002\u0002\u0002\u0005\u0003\u0002\u0002\u0002\u0002",
    "\u0007\u0003\u0002\u0002\u0002\u0002\t\u0003\u0002\u0002\u0002\u0002",
    "\u000b\u0003\u0002\u0002\u0002\u0002\r\u0003\u0002\u0002\u0002\u0002",
    "\u000f\u0003\u0002\u0002\u0002\u0002\u0011\u0003\u0002\u0002\u0002\u0002",
    "\u0013\u0003\u0002\u0002\u0002\u0002\u0015\u0003\u0002\u0002\u0002\u0002",
    "\u0017\u0003\u0002\u0002\u0002\u0002\u0019\u0003\u0002\u0002\u0002\u0002",
    "\u001b\u0003\u0002\u0002\u0002\u0002\u001d\u0003\u0002\u0002\u0002\u0002",
    "\u001f\u0003\u0002\u0002\u0002\u0003!\u0003\u0002\u0002\u0002\u0005",
    "#\u0003\u0002\u0002\u0002\u0007%\u0003\u0002\u0002\u0002\t\'\u0003\u0002",
    "\u0002\u0002\u000b)\u0003\u0002\u0002\u0002\r+\u0003\u0002\u0002\u0002",
    "\u000f-\u0003\u0002\u0002\u0002\u00110\u0003\u0002\u0002\u0002\u0013",
    "3\u0003\u0002\u0002\u0002\u00156\u0003\u0002\u0002\u0002\u00179\u0003",
    "\u0002\u0002\u0002\u0019;\u0003\u0002\u0002\u0002\u001b=\u0003\u0002",
    "\u0002\u0002\u001dE\u0003\u0002\u0002\u0002\u001fI\u0003\u0002\u0002",
    "\u0002!\"\u0007`\u0002\u0002\"\u0004\u0003\u0002\u0002\u0002#$\u0007",
    ",\u0002\u0002$\u0006\u0003\u0002\u0002\u0002%&\u00071\u0002\u0002&\b",
    "\u0003\u0002\u0002\u0002\'(\u0007\'\u0002\u0002(\n\u0003\u0002\u0002",
    "\u0002)*\u0007-\u0002\u0002*\f\u0003\u0002\u0002\u0002+,\u0007/\u0002",
    "\u0002,\u000e\u0003\u0002\u0002\u0002-.\u0007?\u0002\u0002./\u0007?",
    "\u0002\u0002/\u0010\u0003\u0002\u0002\u000201\u0007#\u0002\u000212\u0007",
    "?\u0002\u00022\u0012\u0003\u0002\u0002\u000234\u0007(\u0002\u000245",
    "\u0007(\u0002\u00025\u0014\u0003\u0002\u0002\u000267\u0007~\u0002\u0002",
    "78\u0007~\u0002\u00028\u0016\u0003\u0002\u0002\u00029:\u0007A\u0002",
    "\u0002:\u0018\u0003\u0002\u0002\u0002;<\u0007<\u0002\u0002<\u001a\u0003",
    "\u0002\u0002\u0002=A\t\u0002\u0002\u0002>@\t\u0003\u0002\u0002?>\u0003",
    "\u0002\u0002\u0002@C\u0003\u0002\u0002\u0002A?\u0003\u0002\u0002\u0002",
    "AB\u0003\u0002\u0002\u0002B\u001c\u0003\u0002\u0002\u0002CA\u0003\u0002",
    "\u0002\u0002DF\t\u0004\u0002\u0002ED\u0003\u0002\u0002\u0002FG\u0003",
    "\u0002\u0002\u0002GE\u0003\u0002\u0002\u0002GH\u0003\u0002\u0002\u0002",
    "H\u001e\u0003\u0002\u0002\u0002IJ\t\u0005\u0002\u0002JK\u0003\u0002",
    "\u0002\u0002KL\b\u0010\u0002\u0002L \u0003\u0002\u0002\u0002\u0005\u0002",
    "AG\u0003\u0002\u0003\u0002"].join("");


var atn = new antlr4.atn.ATNDeserializer().deserialize(serializedATN);

var decisionsToDFA = atn.decisionToState.map( function(ds, index) { return new antlr4.dfa.DFA(ds, index); });

function LeftRecursionGrammarLexer(input) {
	antlr4.Lexer.call(this, input);
    this._interp = new antlr4.atn.LexerATNSimulator(this, atn, decisionsToDFA, new antlr4.PredictionContextCache());
    return this;
}

LeftRecursionGrammarLexer.prototype = Object.create(antlr4.Lexer.prototype);
LeftRecursionGrammarLexer.prototype.constructor = LeftRecursionGrammarLexer;

Object.defineProperty(LeftRecursionGrammarLexer.prototype, "atn", {
        get : function() {
                return atn;
        }
});

LeftRecursionGrammarLexer.EOF = antlr4.Token.EOF;
LeftRecursionGrammarLexer.CARET = 1;
LeftRecursionGrammarLexer.STAR = 2;
LeftRecursionGrammarLexer.SLASH = 3;
LeftRecursionGrammarLexer.PERCENT = 4;
LeftRecursionGrammarLexer.PLUS = 5;
LeftRecursionGrammarLexer.MINUS = 6;
LeftRecursionGrammarLexer.EQUAL = 7;
LeftRecursionGrammarLexer.NOT_EQUAL = 8;
LeftRecursionGrammarLexer.LOGICAL_AND = 9;
LeftRecursionGrammarLexer.LOGICAL_OR = 10;
LeftRecursionGrammarLexer.QUESTION = 11;
LeftRecursionGrammarLexer.COLON = 12;
LeftRecursionGrammarLexer.ID = 13;
LeftRecursionGrammarLexer.DIGIT = 14;
LeftRecursionGrammarLexer.WS = 15;

LeftRecursionGrammarLexer.prototype.channelNames = [ "DEFAULT_TOKEN_CHANNEL", "HIDDEN" ];

LeftRecursionGrammarLexer.prototype.modeNames = [ "DEFAULT_MODE" ];

LeftRecursionGrammarLexer.prototype.literalNames = [ null, "'^'", "'*'", 
                                                     "'/'", "'%'", "'+'", 
                                                     "'-'", "'=='", "'!='", 
                                                     "'&&'", "'||'", "'?'", 
                                                     "':'" ];

LeftRecursionGrammarLexer.prototype.symbolicNames = [ null, "CARET", "STAR", 
                                                      "SLASH", "PERCENT", 
                                                      "PLUS", "MINUS", "EQUAL", 
                                                      "NOT_EQUAL", "LOGICAL_AND", 
                                                      "LOGICAL_OR", "QUESTION", 
                                                      "COLON", "ID", "DIGIT", 
                                                      "WS" ];

LeftRecursionGrammarLexer.prototype.ruleNames = [ "CARET", "STAR", "SLASH", 
                                                  "PERCENT", "PLUS", "MINUS", 
                                                  "EQUAL", "NOT_EQUAL", 
                                                  "LOGICAL_AND", "LOGICAL_OR", 
                                                  "QUESTION", "COLON", "ID", 
                                                  "DIGIT", "WS" ];

LeftRecursionGrammarLexer.prototype.grammarFileName = "LeftRecursionGrammar.g4";


exports.LeftRecursionGrammarLexer = LeftRecursionGrammarLexer;

